src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/Target_ARM/ARMReadyToRunGenericHelperNode.cs

```diff
@@ -138,33 +133,20 @@ protected sealed override void EmitCode(NodeFactory factory, ref ARMEmitter enco
 
                             int cctorContextSize = NonGCStaticsNode.GetClassConstructorContextStorageSize(factory.Target, target);
                             encoder.EmitSUB(encoder.TargetRegister.Arg0, ((byte)(cctorContextSize)));
-                            // If we get the required state it will be necessary to restore the original value before returning
-                            encoder.EmitPUSH(encoder.TargetRegister.Arg0);
-
                             // cmp [r0 + ptrSize], 1
-                            encoder.EmitLDR(encoder.TargetRegister.Arg0, encoder.TargetRegister.Arg0, ((short)factory.Target.PointerSize));
-                            encoder.EmitCMP(encoder.TargetRegister.Arg0, ((byte)1));
-                            // return with restoring register
-                            encoder.EmitRETIfEqualPOP(encoder.TargetRegister.Arg0);
-
-                            // just restore after cmp
-                            encoder.EmitPOP(encoder.TargetRegister.Arg0);
+                            encoder.EmitLDR(encoder.TargetRegister.Arg2, encoder.TargetRegister.Arg0, ((short)factory.Target.PointerSize));
```

Comment:
The same fix should be made in ReadyToRunHelperId.GetGCStaticBase.
EmitPOP and EmitRETIfEqualPOP can be deleted because of they will become unused.